files/class-vip-filesystem.php

```diff
@@ -461,4 +463,24 @@ public function filter_wp_read_image_metadata( $meta, $file ) {
 
 		return $meta;
 	}
+
+	/**
+	 * Changes the Font Library directory to work with the VIP Filesystem.
+	 */
+	public function filter_wp_font_dir( $defaults ) {
+		// Without removing the filter, there will be an infinite loop, because
+		// calling wp_upload_dir will trigger upload_dir, which includes wp_get_font_dir
+		// in the context of uploading fonts. That ultimately calls this function
+		// again through the font_dir filter.
+		remove_filter( 'upload_dir', 'wp_get_font_dir' );
+		$upload_dir = wp_upload_dir();
```

Comment:
May be worth switching to `doing_filter` instead, though that may get rolled into core to fix the infinite loop (see https://wordpress.slack.com/archives/C02RQBWTW/p1709082902708339?thread_ts=1708987810.984969&cid=C02RQBWTW)

Reply:
Actually, this infinite loop bug will be fixed here: https://github.com/WordPress/gutenberg/pull/58839